### PR TITLE
Complex json deserialization

### PR DESCRIPTION
**Pull request by mkatychev**
https://github.com/Bestowinc/cogs/pull/6 should be reviewed and merged first

* Added read type `json{}` to signal complex json file traversal: `var4= {path = [[], "nested"], type = "json{}"}`
* Added read type `whole` [when an entire file's contents](https://github.com/Bestowinc/cogs/blob/6e8d29f7985a31f51e51fe56db369a9f8804946d/advanced.cog.toml#L23) should be bound to a key
* This explicit override attempts to safeguard other config retrieval into being flat k/v pairs
* Added `ComplexValue interface{}` to `Cfg` object

### CODE REVIEW

File: cmd/cogs/optparse.go [hunk 1]
```diff
@@ -17,28 +17,30 @@ func ifErr(err error) {
...
 }
 
-func upperKeys(cfgMap map[string]string) map[string]string {
+// upperKeys should always return a flat associative array of strings
+// coercing any interface{} value into a string
+func upperKeys(cfgMap map[string]interface{}) map[string]string {
 	newCfgMap := make(map[string]string)
 	for k, v := range cfgMap {
-		newCfgMap[strings.ToUpper(k)] = v
+		newCfgMap[strings.ToUpper(k)] = fmt.Sprintf("%s", v)
```
Comment (mkatychev) [inline cmd/cogs/optparse.go:35]: I wonder if there's a better pattern for coercing an `interface{}` into a `string` 🤔 ...

**Review by jproffitt — APPROVED** [review]
LGTM

File: file.go [hunk 2]
```diff
@@ -15,29 +15,39 @@ import (
 type readType string
 
 const (
-	readDotenv readType = "dotenv"
-	readJson   readType = "json"
-	deferred   readType = "" // defer file config type to filename suffix
+	rDotenv      readType = "dotenv"
+	rJSON        readType = "json"
+	rJSONComplex readType = "json{}" // complex json key value pair: {"k":{"v1":[],"v2":[]}}
+	rWhole       readType = "whole"  // indicates to associate the entirety of a file to the given key name
```
Comment (jsgoyette) [inline file.go:21]: `whole` is good. But given that it essentially means to not parse the file, might it better be expressed as `raw`?

**Review by jsgoyette — APPROVED** [review]
⭐

File: file.go (hunk 2 above)
Comment (mkatychev) [reply file.go:21]: * After talking this over, I mentioned that the entire file is infact parsed
* A JSON file would be stored and used as a `yaml.Node` 
* Any read JSON file must first be read and only then convered to a `yaml.Node` obj before it’s sent/output back into JSON
https://github.com/Bestowinc/cogs/blob/1018473ac133f14c0c1edefaaf1671dd4de0a092/file.go#L155-L161

**Pull request merged**